File: src/main/java/com/company/erp/features/devices/controller/DeviceController.java
package com.company.erp.features.devices.controller;

import com.company.erp.core.security.AuthenticatedUser;
import com.company.erp.features.devices.dto.DeviceDto;
import com.company.erp.features.devices.dto.RegisterDeviceRequest;
import com.company.erp.features.devices.service.DeviceService;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/me/devices")
public class DeviceController {

    private final DeviceService devices;

    public DeviceController(DeviceService devices) {
        this.devices = devices;
    }

    /** List my registered devices. */
    @GetMapping
    public List<DeviceDto> list() {
        Long me = AuthenticatedUser.require().userId();
        return devices.listForUser(me).stream().map(DeviceDto::from).toList();
    }

    /** Register or update this device's FCM token. Upserts on (userId, deviceId). */
    @PostMapping
    public DeviceDto register(@Valid @RequestBody RegisterDeviceRequest body) {
        Long me = AuthenticatedUser.require().userId();
        return DeviceDto.from(devices.register(me, body));
    }

    /** Revoke this device on logout or token rotation. */
    @DeleteMapping("/{deviceId}")
    public void delete(@PathVariable String deviceId) {
        Long me = AuthenticatedUser.require().userId();
        devices.delete(me, deviceId);
    }
}

File: app/Features/Devices/Controllers/DeviceController.php
<?php

namespace App\Features\Devices\Controllers;

use App\Features\Devices\Dto\DeviceDto;
use App\Features\Devices\Requests\RegisterDeviceRequest;
use App\Features\Devices\Services\DeviceService;
use App\Http\Controllers\Controller;
use Illuminate\Support\Facades\Auth;

class DeviceController extends Controller
{
    /** @var DeviceService */
    private $devices;

    public function __construct(DeviceService $devices)
    {
        $this->devices = $devices;
    }

    /** List my registered devices. */
    public function index()
    {
        $me = (int) Auth::guard('api')->id();
        return $this->devices->listForUser($me)->map([DeviceDto::class, 'from'])->values()->all();
    }

    /** Register or update this device's FCM token. Upserts on (userId, deviceId). */
    public function register(RegisterDeviceRequest $request)
    {
        $me = (int) Auth::guard('api')->id();
        return DeviceDto::from($this->devices->register($me, $request->validated()));
    }

    /** Revoke this device on logout or token rotation. */
    public function destroy(string $deviceId)
    {
        $me = (int) Auth::guard('api')->id();
        $this->devices->delete($me, $deviceId);
        return null;
    }
}
